### PR TITLE
Convert the month widget to exend Month

### PR DESCRIPTION
**Pull request by mfairchild365**
This changes the MonthWidget to extend the Month class, thus reusing logic.

While working on this, it became apparent that we needed better ways to generate URLs.  So a lot of URL related changes are packaged with this.

### CODE REVIEW

File: src/UNL/UCBCN/Frontend/Calendar.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+namespace UNL\UCBCN\Frontend;
+
+class Calendar extends \UNL\UCBCN\Calendar
+{
+    public function getURL()
+    {
+        return Controller::$url . $this->shortname . '/';
```
Comment (saltybeagle) [inline src/UNL/UCBCN/Frontend/Calendar.php]: Any need to urlencode the shortname here?

File: src/UNL/UCBCN/Frontend/Exception.php [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?php
 namespace UNL\UCBCN\Frontend;
 
-interface Exception
+class Exception extends \Exception
```
Comment (saltybeagle) [inline src/UNL/UCBCN/Frontend/Exception.php]: The interface here was intentional. The idea is that we have a base exception interface which all exceptions must implement. When we throw an exception, we use the SPL Exception classes and throw appropriate exceptions. See this doc for more details on why this is important: https://wiki.php.net/pear/rfc/pear2_exception_policy

File: src/UNL/UCBCN/Frontend/Month.php [hunk 3]
```diff
@@ -54,10 +54,12 @@ class Month extends \IteratorIterator
      */
     public function __construct($options)
     {
-        if (isset($options['calendar'])) {
-            $this->calendar = $options['calendar'];
+        if (!isset($options['calendar'])) {
+            throw new Exception('A calendar must be set', 500);
```
Comment (saltybeagle) [inline src/UNL/UCBCN/Frontend/Month.php]: Maybe BadMethodCallException or InvalidArgumentException would be more appropriate here?

File: src/UNL/UCBCN/Frontend/Exception.php (hunk 2 above)
Comment (mfairchild365) [inline src/UNL/UCBCN/Frontend/Exception.php]: fixed in https://github.com/mfairchild365/UNL_UCBCN_Frontend/commit/fb87010cc3092b107b8e5f2e55722304a8c74674

File: src/UNL/UCBCN/Frontend/Month.php (hunk 3 above)
Comment (mfairchild365) [inline src/UNL/UCBCN/Frontend/Month.php]: fixed in https://github.com/mfairchild365/UNL_UCBCN_Frontend/commit/955ffbc920ccea31ed0cb1f1febbd382bbd8309a

File: src/UNL/UCBCN/Frontend/Calendar.php (hunk 1 above)
Comment (mfairchild365) [inline src/UNL/UCBCN/Frontend/Calendar.php]: fixed in https://github.com/mfairchild365/UNL_UCBCN_Frontend/commit/848bb71ffef9aa871ebf29aa821167bf9495c3ac

**Comment by mfairchild365** [comment]
@saltybeagle Ready for another review

**Pull request merged**